tests/models/plants/conftest.py

```diff
@@ -161,10 +187,24 @@ def fixture_canopy_layer_data():
                     [1.76405508e00, 1.76443550e00, 1.72228517e00, 1.14824589e-04],
                     [1.76388228e00, 1.75668428e00, np.nan, np.nan],
                     [1.73664971e00, np.nan, np.nan, np.nan],
+                    [0.98, 0.98, 0.98, 0.98],
+                ]
+            ),
+            [1, 2, 3, 4, 11],
+            # index_filled_canopy, index_surface
+        ),
+        "layer_fapar_canopy_only": (
+            "layer_fapar",
+            np.array(
+                [
+                    [5.86036011e-01, 5.86033790e-01, 5.86046818e-01, 5.93428098e-01],
```

Comment:
What is the first layer here, are we not assuming 3 filled canopy layers? With vegetation in the subcanopy, the indexing will get awkward, something to revisit in the future.

Reply:
Is the below_canopy index now part of index filled canopy?
I think it was the layer_heights_canopy variable that confused me in the first place, why do you have 4 canopy layer heights ? Also the shortwave absorption seems to have too many entries, 3 canopy layers + subcanopy + topsoil =5 not 6? Or is there some vegetation above the first canopy layer?

Reply:
Yup - these test values are tricky because I'm using them to test recalculation at different stages so layers drop in and out, but there _are_ four canopy layers in the first cell. So we have:

above_canopy: always empty
canopy_1 to canopy_4: contain LAI + absorbance data in at least one cell
canopy_6 to canopy_9: contain no data
surface: contains LAI and absorbance
ground: contains absorbance